lib/indexer.rb

```diff
@@ -0,0 +1,121 @@
+module Argo
+  class Indexer
+    SimpleLogJob = Struct.new(:log_val) do
+      def perform
+        sleep(rand 5)
+        job_logger = Logger.new(File.open("tmp/simple_job.log"))
+        job_logger.unknown "#{log_val}"
+      end
+    end
+
+
+    def reindex_object obj
+      doc = obj.to_solr
+      Dor::SearchService.solr.add(doc)
+    end
+
+    def reindex_pid pid
+      obj = Dor.load_instance pid
+      reindex_object obj
+    end
+
+    def reindex_pid_list pid_list, should_commit=false
+      pid_list.each do |pid|
+        reindex_pid pid
+      end
+      ActiveFedora.solr.conn.commit if should_commit
+    end
+
+    def get_pids_for_model_type model_type
+      Dor::SearchService.risearch "select $object from <#ri> where $object <fedora-model:hasModel> #{model_type}"
+    end
+
+
+
+    def reindex_all args
+      index_log = Logger.new(File.join(Rails.root, 'log', 'reindex.log'))
+      index_log.formatter = Logger::Formatter.new
+      index_log.level = ENV['LOG_LEVEL'] ? Logger::SEV_LABEL.index(ENV['LOG_LEVEL']) : Logger::INFO
+      $stdout.sync = true
+      start_time = Time.now
+
+
```

Comment:
i would vote for turning off this rule in hound and rubocop.  more often than not, when i have two lines of whitespace in a row, it's an intentional formatting choice (which i think is reasonable if its judicious).  and it's pretty rare that i commit this sort of thing on accident.

a non-issue in this specific case since this is messy experimentation code, but thought i'd mention it for consideration by others.